experimenting with computational improvement (#7)
whong92/deep-ls: DBUG.md

# Big changes that need to incrementally tested:

> - enumerate_all_tours_edges method (can be tested indepedently, API identical)
> - enumerate_tour_edges method (can be tested indepedently, API identical)
> - caching of tour_adj (can be tested indepedently, API identical)
> - moving the processing of nbh's into envs (first moves and second)
> - vectorizing of nbhs types (and reconstruction)
> - vectorizing of states (and associated changes)
> - grouping of multiple envs into each proc

Author: ```suggestion
- enumerate_all_tours_edges method (can be tested indepedently, API identical)
- enumerate_tour_edges method (can be tested indepedently, API identical)
- caching of tour_adj (can be tested indepedently, API identical)
- moving the processing of nbh's into envs (first moves and second)
- vectorizing of nbhs types (and reconstruction)
- vectorizing of states (and associated changes)
- grouping of multiple envs into each proc
- NB need to maintain parity with multi file env for all the above changes - check always!
```

> - vectorizing of nbhs types (and reconstruction)  --  needs to be done after moving processing into nbh's?
> -- can be done by still passing states together with vectorized state.

Author: ```suggestion
- vectorizing of nbhs types (and reconstruction)
- vectorizing of states (and associated changes)
```

# Big changes that need to incrementally tested:
- enumerate_all_tours_edges method (can be tested indepedently, API identical)
- enumerate_tour_edges method (can be tested indepedently, API identical)
- caching of tour_adj (can be tested indepedently, API identical)
- moving the processing of nbh's into envs (first moves and second)
- vectorizing of nbhs types (and reconstruction)
- vectorizing of states (and associated changes)
- grouping of multiple envs into each proc

> - NB need to maintain parity with multi file env for all the above changes - check always!

Author: multiple instances per env still seems to be a problem
